# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/GamepadHapticActuator.h
# Commit: ec6a64b639a8
# Full Hash: ec6a64b639a82e957376c09a71aef7ce62b3a5aa
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-12-03 09:47:26
# Regressor Bug: 1657404
# File Overlap Count: 1
# Description:
#   Bug 1657404 - Implement a strongly-typed, service-dependent gamepad handle r=handyman,aklotz
#   
#   Currently, the gamepad code uses a uint32_t as a handle and does some trickery
#   with it by trying to create a unique ID and adding an offset to it for VR code.
#   
# ==============================================================================

diff -r 6792f28b99bd -r ec6a64b639a8 dom/gamepad/GamepadHapticActuator.h
--- a/dom/gamepad/GamepadHapticActuator.h	Wed Dec 02 22:36:35 2020 +0000
+++ b/dom/gamepad/GamepadHapticActuator.h	Wed Dec 02 23:06:05 2020 +0000
@@ -11,6 +11,7 @@
 #include "nsWrapperCache.h"
 #include "mozilla/dom/GamepadHapticActuatorBinding.h"
 #include "mozilla/dom/Gamepad.h"
+#include "mozilla/dom/GamepadHandle.h"
 
 namespace mozilla {
 namespace dom {
@@ -18,7 +19,7 @@
 
 class GamepadHapticActuator : public nsISupports, public nsWrapperCache {
  public:
-  GamepadHapticActuator(nsISupports* aParent, uint32_t aGamepadId,
+  GamepadHapticActuator(nsISupports* aParent, GamepadHandle aGamepadHandle,
                         uint32_t aIndex);
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
@@ -41,7 +42,7 @@
 
  protected:
   nsCOMPtr<nsISupports> mParent;
-  uint32_t mGamepadId;
+  GamepadHandle mGamepadHandle;
   GamepadHapticActuatorType mType;
   uint32_t mIndex;
 };